tests/php/test-class-amp-setup-wizard-submenu-page.php

```diff
@@ -0,0 +1,194 @@
+
+	/**
+	 * Setup.
+	 *
+	 * @inheritdoc
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->old_wp_scripts = isset( $GLOBALS['wp_scripts'] ) ? $GLOBALS['wp_scripts'] : null;
+		remove_action( 'wp_default_scripts', 'wp_default_scripts' );
+		remove_action( 'wp_default_scripts', 'wp_default_packages' );
+		$GLOBALS['wp_scripts'] = new WP_Scripts();
+
+		$this->page = new AMP_Setup_Wizard_Submenu_Page( 'amp-options' );
+	}
+
+	/**
+	 * Tear down.
+	 *
+	 * @inheritdoc
+	 */
+	public function tearDown() {
+		parent::tearDown();
+
+		$GLOBALS['wp_scripts'] = $this->old_wp_scripts;
+		add_action( 'wp_default_scripts', 'wp_default_scripts' );
+	}
+
+	/**
+	 * Tests AMP_Setup_Wizard_Submenu_Page::init
+	 *
+	 * @covers AMP_Setup_Wizard_Submenu_Page::init
+	 */
+	public function test_init() {
+		$this->page->init();
+
+		$this->assertEquals( 10, has_action( 'admin_enqueue_scripts', [ $this->page, 'override_scripts' ] ) );
+		$this->assertEquals( 10, has_action( 'admin_enqueue_scripts', [ $this->page, 'enqueue_assets' ] ) );
+	}
+
+	/**
+	 * Tests AMP_Setup_Wizard_Submenu_Page::render
+	 *
+	 * @covers AMP_Setup_Wizard_Submenu_Page::render
+	 */
+	public function test_render() {
+		ob_start();
+
+		$this->page->render();
+
+		$this->assertEquals( trim( ob_get_clean() ), '<div id="amp-setup"></div>' );
+	}
+
+	/**
+	 * Tests AMP_Setup_Wizard_Submenu_Page::screen_handle
+	 *
+	 * @covers AMP_Setup_Wizard_Submenu_Page::screen_handle
+	 */
+	public function test_screen_handle() {
+		$this->assertEquals( $this->page->screen_handle(), 'amp_page_amp-setup' );
+	}
+
+	/**
+	 * Provides test data for test_add_setup_script.
+	 *
+	 * @return array
+	 */
+	public function get_test_setup_scripts() {
+		return [
+			[
+				'asset-1',
+				false,
+			],
+			[
+				'asset-2',
+				true,
+			],
+		];
+	}
+
+	/**
+	 * Tests AMP_Setup_Wizard_Submenu_Page::add_setup_script
+	 *
+	 * @covers AMP_Setup_Wizard_Submenu_Page::add_setup_script
+	 *
+	 * @dataProvider get_test_setup_scripts
+	 *
+	 * @param string  $handle   Script handle
+	 * @param boolean $enqueued Whether to enqueue the script.
+	 */
+	public function test_add_setup_script( $handle, $enqueued ) {
+		$filter_asset = function( $asset, $asset_handle ) use ( $handle ) {
+			if ( $handle !== $asset_handle ) {
+				return $asset;
+			}
+
+			return [
+				'dependencies' => [],
+				'version'      => '1.0',
+			];
+		};
+
+		add_filter( 'amp_setup_asset', $filter_asset, 10, 2 );
+		$this->page->add_setup_script( $handle, $enqueued );
+		remove_filter( 'amp_setup_asset', $filter_asset );
+
+		$this->assertTrue( wp_script_is( $handle, $enqueued ? 'enqueued' : 'registered' ) );
+	}
+
+	/**
+	 * Tests AMP_Setup_Wizard_Submenu_Page::get_asset
+	 *
+	 * @covers AMP_Setup_Wizard_Submenu_Page::get_asset
+	 */
+	public function test_get_asset() {
+		$test_data = [
+			'dependencies' => [],
+			'version'      => '1.0',
+		];
+
+		$filter_asset = function() use ( $test_data ) {
```

Comment:
Closure should be static as it doesn't need instanced data:
```suggestion
		$filter_asset = static function() use ( $test_data ) {
```